l10n_it_ricevute_bancarie/models/riba.py

```diff
@@ -0,0 +1,475 @@
+            list.state = 'draft'
+
+
+class riba_list_line(models.Model):
+    # TODO estendere la account_due_list per visualizzare e filtrare
+    # in base alle riba ?
+    _name = 'riba.list.line'
+    _description = 'Riba details'
+    _rec_name = 'sequence'
+
+    @api.multi
+    def _get_line_values(self):
+        # res = {}
+        for line in self:
+            # res[line.id] = {}
+            line.amount = 0.0
+            line.invoice_date = ""
+            line.invoice_number = ""
+            for move_line in line.move_line_ids:
+                line.amount += move_line.amount
+                if not line.invoice_date:
+                    line.invoice_date = str(
+                        move_line.move_line_id.invoice.date_invoice)
+                else:
+                    line.invoice_date = "%s, %s" % (line.invoice_date, str(
+                        move_line.move_line_id.invoice.date_invoice))
+                if not line.invoice_number:
+                    line.invoice_number = str(
+                        move_line.move_line_id.invoice.internal_number)
+                else:
+                    line.invoice_number = "%s, %s" % (line.invoice_number, str(
+                        move_line.move_line_id.invoice.internal_number))
+
+    amount = fields.Float(
+        compute='_get_line_values', string="Amount")
+    invoice_date = fields.Char(
+        compute='_get_line_values', string="Invoice Date", size=256)
+    invoice_number = fields.Char(
+        compute='_get_line_values', string="Invoice Number", size=256)
+
+    @api.multi
+    def _reconciled(self):  # , cr, uid, ids, name, args, context=None):
+        # wf_service = netsvc.LocalService("workflow")
+        res = {}
+        for line in self:
+            res[line.id] = line.test_paid()
+            if res[line.id]:
+                line.state = 'paid'
+                workflow.trg_validate(
+                    self.env.user.id, 'riba.list',
+                    line.list_id.id, 'paid', cr)
+        return res
+
+    @api.one
+    def move_line_id_payment_gets(self):  # , cr, uid, ids, *args):
+        res = {}
+        if not self:
+            return res
+        cr.execute('SELECT list_line.id, l.id '
+                   'FROM account_move_line l '
+                   'LEFT JOIN riba_list_line list_line ON '
+                   '(list_line.acceptance_move_id=l.move_id) '
+                   'WHERE list_line.id IN %s '
+                   'AND l.account_id=list_line.acceptance_account_id',
+                   (tuple([line.id for line in self]),))
+        for r in cr.fetchall():
+            res.setdefault(r[0], [])
+            res[r[0]].append(r[1])
+        return res
+
+    # return the ids of the move lines which has the same account than the
+    # statement whose id is in ids
+    @api.multi
+    def move_line_id_payment_get(self):  # , cr, uid, ids, *args):
+        if not self:
+            return []
+        ids = [x.id for x in self]
```

Comment:
Isn't this just self.ids? https://www.odoo.com/documentation/8.0/reference/orm.html#model-reference (section recordset operations)